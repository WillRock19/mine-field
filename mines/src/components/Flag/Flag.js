import React from 'react';
import {View} from 'react-native';

import styles from './style';

export default props => {
  return (
    <View style={styles.container}>
      <View style={styles.flag} />
      <View style={styles.flagpole} />
      <View style={styles.smallerBase} />
      <View style={styles.largerBase} />
    </View>
  );
};
