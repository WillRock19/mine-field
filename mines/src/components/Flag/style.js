import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    marginTop: 2,
  },
  flagpole: {
    backgroundColor: '#222',
    height: 14,
    width: 2,
    marginLeft: 9,
    position: 'absolute',
  },
  flag: {
    backgroundColor: '#F22',
    height: 5,
    width: 6,
    marginLeft: 3,
    position: 'absolute',
  },
  smallerBase: {
    backgroundColor: '#F22',
    height: 2,
    width: 6,
    marginLeft: 7,
    marginTop: 10,
    position: 'absolute',
  },
  largerBase: {
    backgroundColor: '#F22',
    height: 2,
    width: 10,
    marginLeft: 5,
    marginTop: 12,
    position: 'absolute',
  },
});
