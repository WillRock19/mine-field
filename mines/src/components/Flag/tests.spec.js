import 'jsdom-global/register';
import 'react-native';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, {shallow, mount} from 'enzyme';
import Flag from './Flag';

Enzyme.configure({adapter: new Adapter()});

describe('Component Flag', () => {
  it('should be rendered correctly', () => {
    const wrapper = shallow(<Flag />);
    expect(wrapper.exists()).toBeTruthy();
  });

  it.only('should have expected structure and styling', () => {
    const wrapper = mount(<Flag />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
