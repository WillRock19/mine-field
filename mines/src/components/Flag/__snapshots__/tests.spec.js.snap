// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Flag should have expected structure and styling 1`] = `
"<_default>
  <View style={{...}}>
    <View style={{...}}>
      <View style={{...}}>
        <View style={{...}} />
      </View>
      <View style={{...}}>
        <View style={{...}} />
      </View>
      <View style={{...}}>
        <View style={{...}} />
      </View>
      <View style={{...}}>
        <View style={{...}} />
      </View>
    </View>
  </View>
</_default>"
`;
